geode-core/src/distributedTest/java/org/apache/geode/internal/cache/persistence/PersistentRecoveryOrderDUnitTest.java

```diff
@@ -1320,6 +1320,7 @@ public void testRecoverAfterConflict() {
           createReplicateRegion(regionName, getDiskDirs(getVMId()));
         });
         assertThat(thrown).isInstanceOf(ConflictingPersistentDataException.class);
+        InternalDistributedSystem.getAnyInstance().getCache().close();
```

Comment:
Can you change this to: getCache().close() like other invokes in this test?

Reply:
There is some difference here. 

`getCache()` will create a new cache. Because `isClosing` is `true` in the old cache, due to `ConflictingPersistentDataException`. In JUnit4CacheTestCase, there is the code:
```
      if (cache == null || cache.isClosed()) {
        cache = null;
        createCache(client, factory);
      }
```

While `InternalDistributedSystem.getAnyInstance().getCache()` returns the old cache. We want to explicitly close the old cache. So we use `InternalDistributedSystem.getAnyInstance().getCache()`.

Reply:
Getting the cache using InternalDistributedSystem will work; but still looks some indirect way of closing the cache....
Can this work:
Get the cache reference; try creating the region and then call close on the cache reference.

vm0.invoke(() -> {
      cache = getCache();
      try (IgnoredException ie = addIgnoredException(ConflictingPersistentDataException.class)) {
        // this should cause a conflict
        Throwable thrown = catchThrowable(() -> {
          createReplicateRegion(regionName, getDiskDirs(getVMId()));
        });
        assertThat(thrown).isInstanceOf(ConflictingPersistentDataException.class);
      }
      cache.close();
    });